### PR TITLE
Move to instantiate-streaming

### LINKED ISSUE #61: Book should use `instantiateStreaming` instead of `compileStreaming`
https://github.com/rust-lang-nursery/rust-wasm/pull/59#issuecomment-365971914

### PR DESCRIPTION
**Pull request by sendilkumarn**
fixes #61 

It is just an idea to make sure that we have covered all the use cases. It is highly unlikely that people when they get started to have a webserver. WDYT?

### CODE REVIEW

**Comment by lukewagner** [comment]
lgtm, thanks!

**Comment by mgattozzi** [comment]
Awesome @sendilkumarn thanks again for helping out with the documentation! :D

File: src/hello-world.md [hunk 1]
```diff
@@ -41,7 +41,27 @@ And we can test it out with:
...
+        });
+    </script>
+  </head>
+  <body></body>
+</html>
+```
+
+Note: To run with `instantiateStreaming` and `compileStreaming`, you need your webserver to serve `.wasm` file with `application/wasm` MIME type.
+
+Alternatively, if you are running locally without any webserver.
```
Comment (est31) [inline src/hello-world.md]: On chromium, the `file://` protocol doesn't allow any XHR requests, which fetch is: https://bugs.chromium.org/p/chromium/issues/detail?id=47416

It works on Firefox though.

Comment (sendilkumarn) [reply src/hello-world.md]: with instantiatestreaming it didnt work in firefox

Comment (mgattozzi) [reply src/hello-world.md]: Is there a Firefox/Chrome agnostic way to do that? We can fix it up then using that.

Comment (est31) [reply src/hello-world.md]: The solution I went with was to base64 encode the wasm into a variable inside a `.js` file and then load that js file together with the other js files via a `<script>` tag, decode that variable inside another file into an array buffer, and to then give that buffer to WebAssembly instantiation functions. This is obviously quite a hack, bloats the size, and requires eval support (which exists per default but it is generally a good security practice to disable it) but from my research this was the only alternative to XHR loading of modules.

It *might* be that one day there is `<script src="file.wasm">` like functionality which would make this easier, but not sure if they will be adopting that or something else. It is listed as one of multiple alternatives in this issue: https://github.com/WebAssembly/design/issues/1087 . If you know any w3c working group members please point this out to them :).

The canonical reply to this Chrome problem that you'll find on the web is "get a web server, get one for web development", and this is probably what most js devs are doing, but if you want to preserve "double click on html file" like functionality for end users, then this is no solution.

Comment (lukewagner) [reply src/hello-world.md]: If I read the tea leaves correctly, ES Module integration will also require an `application/wasm` MIME type; it's really in the same boat as `instantiateStreaming` when it comes to this issue.

Comment (est31) [reply src/hello-world.md]: Oh, that's too bad. Any ideas for the justification? I can't imagine that this is done for security reasons, if `fetch` based instantiation allows you to do it already, but maybe I'm overlooking something.

Comment (lukewagner) [reply src/hello-world.md]: The alternative is content-sniffing the first word to detect JS vs. wasm.  Now the first word of wasm was chosen to start with invalid JS chars, so it's way less worse than other existing forms of content-sniffing on the web, but still people tend to be allergic to content-sniffing.  I think there are some additional subtle issues that I don't understand.

**Pull request merged**